From 56430321795319fc57764dafc0ec05c744c0768d Mon Sep 17 00:00:00 2001
From: "yadi.hu@windriver.com" <yadi.hu@windriver.com>
Date: Tue, 22 Jul 2014 14:15:25 +0800
Subject: [PATCH] ARM: 8031/2: change fixmap mapping region to support 32 CPUs

    a05e54c103b0b8e1dab5d04b411f1d48387c4903 upstream

In 32-bit ARM systems, the fixmap mapping region can support
no more than 14 CPUs(total: 896k; one CPU: 64K). And we can
configure NR_CPUS up to 32. So there is a mismatch. This patch
moves fixmapping region downwards to region 0xffc00000- 0xffe00000.
Then the fixmap mapping region can support up to 32 CPUs.

Reviewed-by: Nicolas Pitre <nico@linaro.org>
Signed-off-by: Liu Hua <sdu.liu@huawei.com>
Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>
Signed-off-by: Hu <yadi.hu@windriver.com>
---
 arch/arm/include/asm/fixmap.h |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm/include/asm/fixmap.h b/arch/arm/include/asm/fixmap.h
index bbae919..13aec4a 100644
--- a/arch/arm/include/asm/fixmap.h
+++ b/arch/arm/include/asm/fixmap.h
@@ -13,8 +13,10 @@
  * 0xfffe0000 and 0xfffeffff.
  */
 
-#define FIXADDR_START		0xfff00000UL
-#define FIXADDR_TOP		0xfffe0000UL
+
+#define FIXADDR_START           0xffc00000UL
+#define FIXADDR_TOP             0xffe00000UL
+
 #define FIXADDR_SIZE		(FIXADDR_TOP - FIXADDR_START)
 
 #define FIX_KMAP_BEGIN		0
-- 
1.7.5.4

